controller/controller.go

```diff
@@ -61,13 +62,13 @@ func NewController(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
 				key, err := cache.MetaNamespaceKeyFunc(obj)
-				if err == nil {
+				if err == nil && hasSubscriptionAnnotation(obj) {
```

Comment:
The applications might get subscriptions via app project or global subscription. This is the first thing that gets checked in `processApp` methods.  https://github.com/argoproj-labs/argocd-notifications/blob/1fc14499c78745dd742587cb0551809678d6256e/controller/controller.go#L171 

If the code is working as expected it should not make any API calls and would take the same amount of time. So optimization in this PR won't help. Does this change improve performance if any way? 

It might be that we just need to increase default number of workers (e.g. 100)

Reply:
Thank you for your review Alex! You are right this is not the right solution. I did try increase --processors-count to 150, but did not see any improvement.